receiver/prometheusreceiver/internal/metricsutil_pdata.go

```diff
@@ -0,0 +1,146 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal"
+
+import "go.opentelemetry.io/collector/model/pdata"
+
+type kv struct {
+	Key, Value string
+}
+
+func distPointPdata(ts pdata.Timestamp, bounds []float64, counts []uint64) *pdata.HistogramDataPoint {
+	hdp := pdata.NewHistogramDataPoint()
+	hdp.SetExplicitBounds(bounds)
+	hdp.SetBucketCounts(counts)
+	hdp.SetTimestamp(ts)
+	var sum float64
+	var count uint64
+	for i, bcount := range counts {
+		count += bcount
+		if i > 0 {
+			sum += float64(bcount) * bounds[i-1]
+		}
+	}
+	hdp.SetCount(count)
+	hdp.SetSum(sum)
+
+	return &hdp
+}
+
+func gaugeDistMetricPdata(name string, kvp []*kv, startTs pdata.Timestamp, points ...*pdata.HistogramDataPoint) *pdata.Metric {
+	hMetric := cumulativeDistMetricPdata(name, kvp, startTs, points...)
+	hMetric.Histogram().SetAggregationTemporality(pdata.MetricAggregationTemporalityDelta)
```

Comment:
From what I understand, all prometheus data is cumulative including gauge, but am likely wrong.

Wondering if there is a reference for the mapping, I searched around and found this doc with a broken link to a design doc. Would help reviewing this PR if there's information about the mapping somewhere

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/openmetrics-guidelines.md

Reply:
I'm not actually sure.  This is inherited test helper code that seems to be used to test adjustment of gauge histograms, which should never be encountered since the appender won't build them and the metric adjuster will short circuit when encountering a distribution with other than cumulative temporality.  I think this can be removed without impacting functionality or test coverage.